### ISSUE TITLE
Improve the CUDA out-of-the box experience with the PyTorch preset

### ISSUE THREAD

**Issue by sbrunk** [bytedeco/javacpp-presets#1304]
I've done some experiments with the PyTorch preset to figure out how we could provide a similar experience regarding CUDA as you get with the Python wheels. When you do a `pip install torch` on linux you'll get a version bundled with the CUDA libraries that usually "just works" even if you don't have the CUDA toolkit (in the right version) installed locally. You only need the hardware and a recent enough driver.

Installing `org.bytedeco:pytorch-platform-gpu:1.13.1-1.5.9-SNAPSHOT` from maven on the other hand, requires a user to install the right toolkit, which can be hairy sometimes, and then, depending on the installation method, also set the `LD_LIBRARY_PATH`, or the java library path correspondingly.
The other option is to download the matching LibTorch version which comes bundled with cuda/cudnn etc. like the wheel, and set
`-Dorg.bytedeco.javacpp.pathsFirst=true -Djava.library.path=/path/to/libtorch/lib`

Both options require manual extra effort and kind of defeat the purpose of what makes JavaCPP awesome (among other things): Almost transparently provide native libs as maven dependencies, multiplatform, together with the loading and caching mechanism.

So after trying a few things with repackaging stuff I realized that what I was looking for is already there: The [CUDA presets](https://github.com/bytedeco/javacpp-presets/tree/master/cuda) already provide the native CUDA libs nicely packaged and integrated with the JavaCPP caching and loading mechanisms. :)
So adding `org.bytedeco:cuda-platform-redist:11.8-8.6-1.5.9-SNAPSHOT` (or the corresponding classifier variants) gets us very close to providing everything via maven.

The only library still missing now seems to be `libcupti`, from the [CUDA Profiling Tools Interface](https://developer.nvidia.com/cuda-profiling-tools-interface). If I'm trying to use PyTorch now, I get the following error message:
```
libjnitorch.so: libcupti.so.11.8: cannot open shared object file: No such file or directory
```
Seems to be required by `libtorch`:
```bash
ldd libtorch_cuda.so
...
    libcupti.so.11.8 => not found
...
```

Installing just `libcupti` locally and putting it on the library path makes it work. So I believe that it could be enough to disable linking against it during the libtorch build if possible (upstream libtorch does not seem to link against libcupti), or add it there or to the cuda build.

WDYT? @saudet @HGuillemet

**Comment by saudet** [comment]
If it's missing from the resist artifact, let's add it, yes

**Comment by saudet** [comment]
Or if it's not important enough, we can disable it from the build, sure, but why is it enabled by default then?

**Comment by saudet** [comment]
So according to https://github.com/pytorch/pytorch/issues/74473 it looks like they're either bundling the shared library or linking with it statically. Either option is fine, as long as it works. Please open a pull request with the option that you prefer! Thanks

**Comment by sbrunk** [comment]
It seems like they switched to the shared library approach in https://github.com/pytorch/pytorch/commit/d31635475989102088b7812d52bad51ce4b1a011 due to [issues with static linking](https://github.com/pytorch/pytorch/issues/57744).

**Comment by sbrunk** [comment]
Attempt to bundle libcupti in #1307

**Comment by sbrunk** [comment]
Fixed by #1307
